engine/src/main/java/org/openobservatory/engine/Engine.java

```diff
@@ -2,81 +2,81 @@
 
 import android.content.Context;
 
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.Set;
-import java.util.Vector;
+import java.io.IOException;
 
 /**
  * Engine is a factory class for creating several kinds of tasks. We will use different
  * engines depending on the task that you wish to create.
  */
-public class Engine {
-    private static Set<String> probeEngineTasks = new HashSet<>(Arrays.asList(
-            "Telegram",
-            "Dash",
-            "Ndt",
-            "Psiphon",
-            "Tor",
-            "Whatsapp",
-            "FacebookMessenger",
-            "HttpHeaderFieldManipulation",
-            "HttpInvalidRequestLine",
-            "WebConnectivity"
-    ));
-
-    /** getVersionMK returns the version of Measurement Kit we're using */
-    public static String getVersionMK() {
-        return io.ooni.mk.MKVersion.getVersionMK();
-    }
-
+public final class Engine {
     /** newUUID4 returns the a new UUID4 for this client  */
     public static String newUUID4() {
         return oonimkall.Oonimkall.newUUID4();
     }
 
     /** startExperimentTask starts the experiment described by the provided settings. */
-    public static ExperimentTask startExperimentTask(ExperimentSettings settings) throws EngineException {
-        if (probeEngineTasks.contains(settings.taskName())) {
-            try {
-                return new OONIProbeEngineTaskAdapter(
-                        oonimkall.Oonimkall.startTask(settings.serialization())
-                );
-            } catch (Exception exc) {
-                throw new EngineException("cannot start OONI Probe Engine task", exc);
-            }
+    public static OONIMKTask startExperimentTask(OONIMKTaskConfig settings) throws OONIException {
+        try {
+            return new PEMKTask(
+                    oonimkall.Oonimkall.startTask(settings.serialization())
+            );
+        } catch (Exception exc) {
+            throw new OONIException("cannot start OONI Probe Engine task", exc);
         }
-        return new MKExperimentTaskAdapter(io.ooni.mk.MKAsyncTask.start(settings.serialization()));
     }
 
-    /** maybeUpdateResources updates the bundled resources if needed */
-    public static boolean maybeUpdateResources(Context context) {
-        return io.ooni.mk.MKResourcesManager.maybeUpdateResources(context);
+    /** resolveProbeCC creates a new GeoIP lookup task, runs it and return the probeCC. */
```

Comment:
`creates` => `returns`

Reply:
read the entire description

Reply:
lol - still I think the documentation should not mention implementation details and only say what this function does